### DISCUSSION TITLE
Use component names when schemas point by $ref

### DISCUSSION THREAD

**Discussion by blanquer** [stoplightio/elements#2319] [General]
Hello,

From a usability perspective, "unrolling" "named" schemas inline in the UI feels unnecessarily confusing. It would be much better to try to display the name of the related schema instead (maybe with an option to expand).

More specifically, here are a couple of good examples:
* If you have a response body that returns a `Pet` schema (or a collection of `Pet`s), it would be much nicer to simply display `array of: Pet` rather than `array of: ....a possibly long list structure that matches what a Pet is...`
* If you are displaying a `Pet` schema, and one of the properties refers to an `Owner` schema by $ref, it would also be nice to do the same.

I am unsure if the JsonSchema viewer has any ability to do so, or how the elements UI could be made so that it displays it that way. This could start by simply using the key of the hash for the component (whatever that unique string name is), and later there might be better customization options like an `x-*` property on those, for a better display name if necessary.

I'd appreciate any thoughts or pointers to get that working, as this would make presenting the API docs MUCH easier to read.

**Comment by pmackay** [comment]
I agree with this. Is there currently any way to configure whether nested schemas are expanded/collapsed by default?

**Comment by ptagr** [comment]
yeah. the unrolling makes it confusing and there are details lost. if the same schema is used across objects, there is no easy way to figure it out. and the object names are also not displayed.
